WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsFragment.kt

```diff
@@ -183,15 +184,27 @@ class NewSiteCreationVerticalsFragment : NewSiteCreationBaseFormFragment<NewSite
         viewModel = ViewModelProviders.of(this, viewModelFactory)
                 .get(NewSiteCreationVerticalsViewModel::class.java)
 
-        viewModel.uiState.observe(this, Observer { state ->
-            state?.let {
-                updateVisibility(contentLayout, state.contentState == CONTENT)
-                updateVisibility(fullscreenErrorLayout, state.contentState == FULLSCREEN_ERROR)
-                updateVisibility(fullscreenProgressLayout, state.contentState == FULLSCREEN_PROGRESS)
-                updateVisibility(skipButton, state.showSkipButton)
-                updateHeader(state.headerUiState)
-                updateSearchInput(state.searchInputState)
-                updateSuggestions(state.items)
+        viewModel.uiState.observe(this, Observer { uiState ->
+            uiState?.let {
+                when (uiState) {
```

Comment:
I feel like this is a bit too much logic for the view layer. Can we just add the visibility flags to the base `VerticalsUiState` class instead and simplify this whole thing? Here is an example:

```
sealed class VerticalsUiState(
    fullscreenProgressLayoutVisibility: Boolean,
    contentLayoutVisibility: Boolean,
    fullscreenErrorLayoutVisibility: Boolean
) {
    data class VerticalsContentUiState(
        val searchInputUiState: VerticalsSearchInputUiState,
        val headerUiState: VerticalsHeaderUiState?,
        val showSkipButton: Boolean,
        val items: List<VerticalsListItemUiState>
    ) : VerticalsUiState(
            fullscreenProgressLayoutVisibility = false,
            contentLayoutVisibility = true,
            fullscreenErrorLayoutVisibility = false
    )

    object VerticalsFullscreenProgressUiState : VerticalsUiState(
            fullscreenProgressLayoutVisibility = true,
            contentLayoutVisibility = false,
            fullscreenErrorLayoutVisibility = false
    )
    data class VerticalsFullscreenErrorUiState constructor(
        val titleResId: Int,
        val subtitleResId: Int? = null
    ) : VerticalsUiState(
            fullscreenProgressLayoutVisibility = false,
            contentLayoutVisibility = false,
            fullscreenErrorLayoutVisibility = true
    ) {
        companion object {
            fun createGenericErrorUiState() = VerticalsFullscreenErrorUiState(
                    string.site_creation_error_generic_title,
                    string.site_creation_error_generic_subtitle
            )

            fun createConnectionErrorUiState() =
                    VerticalsFullscreenErrorUiState(R.string.site_creation_error_connection_title)
        }
    }
}
```